docker-images/kafka/scripts/zookeeper_run.sh

```diff
@@ -51,5 +53,34 @@ fi
 
 . ./set_kafka_gc_options.sh
 
+############ ZK Upgrade Start ############
+# ZK 3.4.x -> 3.5.x upgrade adds snapshot checks that can cause ZK pods with persistent storage to fail on startup
+# The section below disables the snapshot checks until the servers have created a snapshot.
+
+# If this is a 3.4.x server then place an indicator file in the data directory so this persists over multiple restarts
```

Comment:
How do you know this will be ever executed before the upgrade?

Reply:
To do the upgrade from a 3.4.x server at some point a 3.4.x server must have run this script at some point? If so then the flag file will be in the persistent storage and the snapshot checks will be done.

Reply:
Is it really? This will be only in the new `0.16.0` release images presumably with Kafka 2.4.0. So I think there will be people jumping from 0.15.0 to 0.16.0 without having a version specified in their resource who will just upgrade from 0.15.0 3.4 Zoo to 0.16.0 3.5 Zoo.

Reply:
Well yes, that will be an issue.

Reply:
Ok to get around this issue I will have to add env var setting in the CO.